AppCenterReactNativeShared/android/src/main/java/com/microsoft/appcenter/reactnative/shared/AppCenterReactNativeShared.java

```diff
@@ -59,25 +82,15 @@ public static synchronized void configureAppCenter(Application application) {
             AppCenterLog.debug(LOG_TAG, "Configure with secret.");
             AppCenter.configure(application, sAppSecret);
         }
-
-        /*
-         * When sStartAutomatically flag is set to true, every service (analytics/auth/crashes/etc.)
-         * will be started by separate AppCenter.start call. If Auth module is used,
-         * call doNotResetAuthAfterStart to avoid resetting the auth token.
-         */
-        try {
-            Class.forName("com.microsoft.appcenter.auth.Auth");
-            AuthTokenContext.getInstance().doNotResetAuthAfterStart();
-        } catch (Exception ignored) {
-
-            /* Nothing to handle; this is reached if Auth SDK isn't used. */
-        }
     }
 
-    private static void readConfigurationFile() {
+    public static synchronized JSONObject readConfigurationFile(Application application) {
+        if (sConfiguration != null) {
+            return sConfiguration;
+        }
         try {
             AppCenterLog.debug(LOG_TAG, "Reading " + APPCENTER_CONFIG_ASSET);
-            InputStream configStream = sApplication.getAssets().open(APPCENTER_CONFIG_ASSET);
+            InputStream configStream = application.getAssets().open(APPCENTER_CONFIG_ASSET);
```

Comment:
why introducing the parameter? we already set the static one

Reply:
We introduced `application` parameter so that `readConfigurationFile` utility method can be safely called in `AppCenterReactNativeModule` class (https://github.com/microsoft/appcenter-sdk-react-native/pull/711/files#diff-a6178579ac3d215e9f3df193c1ce6e60R33) before `configureAppCenter` sets the static `sApplication`. Note thatiIn the past `readConfigurationFile` is a private static method only called inside `configureAppCenter` method, but it's a utility method that can be called outside `configureAppCenter`.

Reply:
You don't need Application in signature then. `Context context` will do.